r/R/dataset.R

```diff
@@ -141,6 +141,22 @@ Dataset$create <- function(children, schema) {
 #' @export
 names.Dataset <- function(x) names(x$schema)
 
+#' @export
+dim.Dataset <- function(x) {
+  rows <- sum(purrr::map_dbl(x$files, ~ParquetFileReader$create(.x)$ReadTable()$num_rows))
+  cols <- x$schema$num_fields
+
+  c(rows, cols)
```

Comment:
Totally agree but since `nrow` and `ncol` aren't generics, it isn't clear to me how to do this without causing issues the base implementation of `ncol`/`nrow`.

Reply:
hmm yeah, you're right, probably not worth masking them. Of course if we were using S4 methods we could, so maybe we should switch 😭 🤣

Reply:
I get to working on that switch right away 🥇